src/components/root/createModals/CreateEnvironmentModal.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+	Autocomplete,
+	Button,
+	DialogActions,
+	DialogContent,
+	DialogTitle,
+	Divider,
+	FormControl,
+	FormLabel,
+	Input,
+	Modal,
+	ModalDialog,
+} from '@mui/joy';
+import { CreateModalsProps } from './createModalsProps';
+import { Environment, iconFromTabType } from '../../../types/application-data/application-data';
+import { useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { selectEnvironments } from '../../../state/active/selectors';
+import { useAppDispatch } from '../../../state/store';
+import { addNewEnvironment } from '../../../state/active/thunks/environments';
+import { addTabs, setSelectedTab } from '../../../state/tabs/slice';
+
+export function CreateEnvironmentModal({ open, closeFunc }: CreateModalsProps) {
+	const [envName, setEnvName] = useState('');
+	const [cloneFrom, setCloneFrom] = useState<string | null>(null);
+	const allEnvironments = useSelector(selectEnvironments);
+	const dispatch = useAppDispatch();
+	const createEnvironmentFunction = async () => {
+		let newEnvironment: Partial<Environment> = { __name: envName };
+		if (cloneFrom != null) {
+			const environmentToCloneFrom = allEnvironments[cloneFrom];
+			if (environmentToCloneFrom != undefined) {
+				newEnvironment = { ...structuredClone(environmentToCloneFrom), ...newEnvironment };
+				delete newEnvironment['__id'];
+			}
+		}
+		const createdEnvironmentId = await dispatch(addNewEnvironment({ data: newEnvironment })).unwrap();
+		dispatch(addTabs({ [createdEnvironmentId]: 'environment' }));
+		dispatch(setSelectedTab(createdEnvironmentId));
+	};
+	const envNameValid = envName.length > 0;
+	const allFieldsValid = useMemo(() => envNameValid, [envName]);
+	return (
+		<Modal
+			open={open}
+			onClose={() => {
+				closeFunc();
+			}}
+		>
+			<ModalDialog variant="outlined" role="alertdialog">
+				<DialogTitle>
+					{iconFromTabType['environment']}
+					Create New Environment
+				</DialogTitle>
+				<Divider />
+				<DialogContent>
+					<FormControl>
+						<FormLabel>Environment Name *</FormLabel>
+						<Input value={envName} onChange={(e) => setEnvName(e.target.value)} error={!envNameValid} required />
+					</FormControl>
+					<FormControl>
+						<FormLabel>Clone from existing environment?</FormLabel>
+						<Autocomplete
+							value={{
+								label: allEnvironments[cloneFrom as string]?.__name ?? "Don't clone",
```

Comment:
Oh I see, since you already select all environments for this, I guess you were aiming to re-use it rather than introduce another selector.